File: C2110I/AP/11-11-2022/myapp/myapp/models/Student.cs
using System;
using System.Xml.Linq;

namespace myapp.models {
	public class Student {
		public string?	Name  { get; set; }
		public string?	Province { get; set; }
		public int		BirthYear { get; set; }
		public Result?	Mark { get; set; }

		public void Input()
		{
			//validate here!
			Console.WriteLine("Enter student's name: ");
			Name = Console.ReadLine() ?? "";

            Console.WriteLine("Enter student's province: ");
            Province = Console.ReadLine() ?? "";

			while (true) {
                Console.WriteLine("Enter student's birthYear: ");
                BirthYear = Convert.ToInt16(Console.ReadLine());
				int age = DateTime.Now.Year - BirthYear;
				bool isValidAge = age > 18 && age < 45;
				if (isValidAge) {
					break;
				}
            }			

			Mark = new Result();
			Mark.Input();
        }
	}
}

File: C2108G1/ADF1/13-09-2022/myapp/src/main/java/org/myapp/nguyenvana/models/Student.java
package org.myapp.nguyenvana.models;

import java.util.Calendar;

public class Student  {
    public Student() {}
    //encapsulation:
    //fieds are "private", method should be "public"
    private String name;
    private String province;
    private int birthYear;
    private Result mark;

    public String getName() {
        return name;
    }

    public Student(String name, String province,
                   int birthYear, Result mark) {
        this.name = name;
        this.province = province;
        this.birthYear = birthYear;
        this.mark = mark;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getProvince() {
        return province;
    }

    public void setProvince(String province) {
        this.province = province;
    }

    public int getBirthYear() {
        return birthYear;
    }
    public int getAge() {
        return Calendar.getInstance().get(Calendar.YEAR) - birthYear;
    }
    public void setBirthYear(int birthYear) {
        this.birthYear = birthYear;
    }


    public Result getMark() {
        return mark;
    }

    public void setMark(Result mark) {
        this.mark = mark;
    }

    @Override
    public String toString() {
        return "Student's detail: \n" +
                "name : " + name + "\n" +
                "province : " + province + "\n" +
                "birthYear : " + birthYear + "\n" +
                "age : " + this.getAge() + "\n" +
                "mark : " + mark + "\n";
    }
}
